backend/fms_core/services/experiment_run_info.py

```diff
@@ -0,0 +1,347 @@
+
+    # individual fields
+    expected_sex: Union[str, None] = None
+    ncbi_taxon_id: Union[int, None] = None
+    taxon_name: Union[str, None] = None
+    
+    # library fields
+    library_type: Union[str, None] = None
+    library_size: Union[float, None] = None
+    index_set_obj_id: Union[str, None] = None
+    index_set_name: Union[str, None] = None
+    index_obj_id: Obj_Id = None
+    index_name: Union[str, None] = None
+    index_sequence_3_prime: Union[List[str], None] = None
+    index_sequence_5_prime: Union[List[str], None] = None
+    library_kit: Union[str, None] = None
+
+    # capture fields
+    capture_kit: Union[str, None] = None
+    capture_baits: Union[str, None] = None
+
+    #ChIP-Seq
+    chip_seq_mark: Union[str, None] = None
+    
+
+@dataclass
+class RunInfo:
+    # Manifest version (1.0.0 to start) 
+    version: str
+
+    # Experiment run name and id
+    run_name: str
+    run_obj_id: int
+    run_start_date: Union[str, None]
+
+    # Flowcell / container for experiment
+    container_obj_id: int
+    container_barcode: str
+
+    platform: str
+    instrument_serial_number: str
+    instrument_type: str
+
+    samples: List[RunInfoSample]
+
+
+RUN_INFO_FILE_VERSION = "1.0.0"
+
+def generate_run_info_file(experiment_run: ExperimentRun):
+    '''
+    Generate the run info for an experiment, to be submitted to tech dev for run processing.
+
+    This function writes the data to the file it is given. The file must be text-based, open,
+    and writable.
+
+    The output is a RunInfo object in the JSON format.
+    '''
+    run_info = generate_run_info(experiment_run)
+
+    file_name_base = experiment_run.name if experiment_run.name else 'experiment_run'
+    file_name = make_timestamped_filename(file_name_base, 'json')
+
+    path = os.path.join(settings.RUN_INFO_OUTPUT_PATH, file_name)
+
+    with open(path, "w", encoding="utf-8") as file:
+        json.dump(run_info, file, indent=4)
+
+def generate_run_info(experiment_run: ExperimentRun) -> RunInfo:
+    ''' 
+    Generates the run info for the experiment, including all of the derived
+    samples in the experiment.
+
+    Returns a RunInfo object as a dict.
+    '''
+    instrument: Instrument = experiment_run.instrument
+
+    start_date = None
+    if experiment_run.start_date is not None:
+        start_date = experiment_run.start_date.strftime("%Y-%m-%d")
+
+    run_info : RunInfo = RunInfo(
+        version=RUN_INFO_FILE_VERSION,
+        run_name=experiment_run.name or '',
+        run_obj_id=experiment_run.pk,
+        run_start_date= start_date,
+        container_obj_id=experiment_run.container.pk,
+        container_barcode=experiment_run.container.barcode,
+        platform=instrument.type.platform.name,
+        instrument_serial_number=experiment_run.instrument.serial_id,
+        instrument_type=instrument.type.type,
+        samples=[]
+    )
+
+    run_info.samples = _generate_run_info_samples(experiment_run)
+
+    return asdict(run_info)
```

Comment:
We should return errors and warnings and propagate them between sub-functions. We expect services to provide information on failures. Here if requesting to start the run processing we expect to have some form of error message if there is failure (the user can forward the message to us in that case)

Reply:
My plan was to have the service catch any exceptions that occur and populate the errors[] with the exception strings.

Reply:
Yeah maybe we should move this code out of services... we can let it here for now, but it is exactly the kind of code i talked about during the summer (neither a service nor anything else...)